dds_web/api/files.py

```diff
@@ -740,3 +740,22 @@ def put(self):
             db.session.commit()
 
         return {"message": "File info updated."}
+
+
+class UpdateFailedFiles(flask_restful.Resource):
+    """Add files from failed_delivery_log to DB using the "add_uploaded_files_to_db" utils function"""
+
+    @auth.login_required(role=["Unit Admin", "Unit Personnel", "Project Owner", "Researcher"])
```

Comment:
Only Unit Admins and Personnel can upload, so only they should be able to use this endpoint
```suggestion
    @auth.login_required(role=["Unit Admin", "Unit Personnel"])
```